karapace/anonymize_schemas/anonymize_avro.py

```diff
@@ -0,0 +1,101 @@
+"""
+karapace - anonymize avro
+
+Copyright (c) 2022 Aiven Ltd
+See LICENSE for details
+"""
+from typing import Any, Dict, List, Union
+
+import hashlib
+
+ALIASES = "aliases"
+DEFAULT = "default"
+DOC = "doc"
+ENUM = "enum"
+FIELDS = "fields"
+ITEMS = "items"
+LOGICAL_TYPE = "logicalType"
+NAME = "name"
+NAMESPACE = "namespace"
+ORDER = "order"
+PRECISION = "precision"
+SCALE = "scale"
+SIZE = "size"
+SYMBOLS = "symbols"
+TYPE = "type"
+
+
+# Doc and order field are special case and not included
+KEYWORDS = [ALIASES, DEFAULT, ENUM, FIELDS, ITEMS, LOGICAL_TYPE, NAME, NAMESPACE, PRECISION, SCALE, SIZE, SYMBOLS, TYPE]
+PRIMITIVE_TYPES = ["null", "boolean", "int", "long", "float", "double", "bytes", "string"]
+LOGICAL_TYPES = [
+    "date",
+    "decimal",
+    "duration",
+    "local-timestamp-micros",
+    "local-timestamp-millis",
+    "time-micros",
+    "time-millis",
+    "timestamp-micros",
+    "timestamp-millis",
+    "uuid",
+]
+ALL_TYPES = PRIMITIVE_TYPES + LOGICAL_TYPES + ["array", ENUM, "fixed", "map", "record"]
+
+ORDER_VALID_VALUES = ["ascending", "descending", "ignore"]
+
+
+def anonymize_name(name: str) -> str:
```

Comment:
This function assumes that names are always valid in the input data.
If there are invalid names the algorithm would need changes:

* Split by dot.
* For each element.
    * Hash the string.
    * Add the first character from the original string to start of the hash.
    * Collect every character from the original string not matching to the Avro name rule and add to the end of the string
* Concatenate elements by dot to single string.

After these steps the anonymized string contains also the invalid characters.